### PR TITLE
Check for Extra pointer dereferencing

### PR DESCRIPTION
**Pull request by jdramani**
Solves issue #7640

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/7643)

### CODE REVIEW

**Comment by jdramani** [comment]
Build failed as expected.

**Comment by jdm** [comment]
@jdramani Want to fix them, too? :)

**Comment by jdm** [comment]
Oh, having looked at these changes, I would rather integrate this check into the existing `check_rust` method, rather than making a new one.

**Comment by Manishearth** [comment]
Doesn't this make more sense as a `check_ty` lint, since there may be cases when we want to override it?

**Comment by jdm** [comment]
@Manishearth: why would we want to override it?

**Comment by Manishearth** [comment]
In case of APIs that handle generic container &Ts which don't have `?Sized` bounds

**Comment by nox** [comment]
```
./components/compositing/compositor.rs:694: use &[T] instead of &Vec<T>
./components/compositing/compositor_layer.rs:106: use &[T] instead of &Vec<T>
./components/compositing/compositor_layer.rs:286: use &[T] instead of &Vec<T>
./components/gfx/display_list/mod.rs:207: use &[T] instead of &Vec<T>
./components/net/resource_task.rs:64: use &[T] instead of &Vec<T>
./components/net/resource_task.rs:71: use &[T] instead of &Vec<T>
./components/style/animation.rs:710: use &[T] instead of &Vec<T>
./components/style/animation.rs:711: use &[T] instead of &Vec<T>
./components/style/animation.rs:737: use &[T] instead of &Vec<T>
./components/style/animation.rs:738: use &[T] instead of &Vec<T>
./components/style/animation.rs:797: use &[T] instead of &Vec<T>
```

**Comment by jdm** [comment]
Almost there! Like @nox pointed out, this can't merge until the existing problems this picks up are fixed, too :)
-S-awaiting-review 

---

Reviewed 1 of 1 files at r1, 1 of 1 files at r2, 1 of 1 files at r3.
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

<sup>**[python/tidy.py, line 310 \[r4\]](https://reviewable.io:443/reviews/servo/servo/7643#-JzVUTuFRZIU4rhpL8CI)** ([raw file](https://github.com/servo/servo/blob/63e795ced94c4cf9fc9bd441c734dbb923121da1/python/tidy.py#L310)):</sup>
This can be `if ": &Vec<" in line:`.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7643)

**Comment by jdm** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7643)

**Comment by jdm** [comment]
-S-awaiting-review -S-fails-tidy +S-needs-code-changes 

---

Reviewed 5 of 5 files at r5, 4 of 4 files at r6.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7643)

**Comment by jdramani** [comment]
Have i done something wrong here? Pls tell me as this is my first time..and sorry for taking this long

**Comment by Manishearth** [comment]
It looks like you merged master when you should have rebased. Try rebasing over master:

```
git checkout master
git pull upstream master
git checkout extra_ptr_dref
git rebase master
```

**Comment by jdramani** [comment]
@Manishearth  Thanks :)

**Comment by Manishearth** [comment]
I rebased it for you, since there was some nontrivial brokenness: https://github.com/servo/servo/compare/master...Manishearth:pr-7643

Run:

```
git checkout master
git fetch git@github.com:Manishearth/servo pr-7643:extra_ptr_dref -f
git checkout extra_ptr_dref
git push -f git@github.com:jdramani/servo extra_ptr_dref
```

**Comment by jdramani** [comment]
@Manishearth The second command does not work. It gives following:
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.


On Tue, Sep 22, 2015 at 3:53 AM, Manish Goregaokar <notifications@github.com

**Comment by Manishearth** [comment]
You don't have ssh set up wit git. No problem.

```
git checkout master
git fetch https://github.com/Manishearth/servo.git pr-7643:extra_ptr_dref -f
git checkout extra_ptr_dref
git push -f https://github.com/github.com:jdramani/servo.git extra_ptr_dref
```

**Comment by jdramani** [comment]
@Manishearth Thanks for the help man :) And out of curiosity, what did i do wrong previously? (Nontrivial brokenness)

**Comment by Manishearth** [comment]
Not exactly sure. You seem to have merged more than once, which duplicated some commits.

**Comment by jdm** [comment]
Sorry, two last requests - first, the rebase isn't quite right, since the `print_items` method was removed from the official master branch. Second, please [squash](https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/) all the commits into one.

**Comment by jdramani** [comment]
Pls let me know if there is still any problem. This being my first PR i am struggling a bit with commands.

**Comment by Manishearth** [comment]
Try copying the commands I gave.

**Comment by jdramani** [comment]
@Manishearth 
These ones,right?
git checkout master
git fetch https://github.com/Manishearth/servo.git pr-7643:extra_ptr_dref -f
git checkout extra_ptr_dref
git push -f https://github.com/github.com/jdramani/servo.git extra_ptr_dref

File: python/tidy.py [hunk 1]
```diff
@@ -109,8 +109,9 @@ def check_by_line(file_name, contents):
         errors = itertools.chain(
             check_length(file_name, idx, line),
             check_whitespace(idx, line),
-            check_whatwg_url(idx, line),
+            check_whatwg_url(idx, line)
```
Comment (frewsxcv) [inline python/tidy.py]: In the future, you should not remove these commas

**Comment by jdramani** [comment]
Pls take a look at this. Tell me if there is any problem

**Comment by jdm** [comment]
Looks great! Thanks for sticking with it :)

**Comment by nox** [comment]
@bors-servo retry

#7422

**Pull request merged**